instrumentation/spring-webmvc-3.1/src/main/java/io/opentelemetry/auto/instrumentation/springwebmvc/HandlerAdapterInstrumentation.java

```diff
@@ -85,7 +86,7 @@ public static SpanWithScope nameResourceAndStartSpan(
       // Name the parent span based on the matching pattern
       final Object parentContext = request.getAttribute(CONTEXT_ATTRIBUTE);
       if (parentContext instanceof Context) {
-        DECORATE.onRequest(getSpan((Context) parentContext), request);
+        DECORATE.updateSpanNameUsingPattern(getSpan((Context) parentContext), request);
```

Comment:
`SpringWebMvcDecorator.DECORATOR` now extends `HttpServerTracerBase`. I had to rename this method to avoid a name conflict. DECORATOR.OnRequest is now used to set span attributes and create the default span name while `DECORATE.updateSpanNameUsingPattern()` updates a span with a custom name.